modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxNdfTradeCsvPlugin.java

```diff
@@ -0,0 +1,157 @@
+
+  private static final ImmutableList<String> HEADERS = ImmutableList.<String>builder()
+      .add(PAYMENT_DATE_FIELD)
+      .add(SETTLEMENT_CURRENCY_FIELD)
+      .add(SETTLEMENT_CURRENCY_DIRECTION_FIELD)
+      .add(SETTLEMENT_CURRENCY_NOTIONAL_FIELD)
+      .add(NON_DELIVERABLE_CURRENCY_FIELD)
+      .add(FX_RATE_FIELD)
+      .add(FX_INDEX_FIELD)
+      .build();
+
+  //-------------------------------------------------------------------------
+  @Override
+  public Set<String> tradeTypeNames() {
+    return ImmutableSet.of("FXNDF", "FX NDF", "NDF");
+  }
+
+  @Override
+  public Optional<Trade> parseTrade(
+      Class<?> requiredJavaType,
+      CsvRow baseRow,
+      List<CsvRow> additionalRows,
+      TradeInfo info,
+      TradeCsvInfoResolver resolver) {
+
+    if (requiredJavaType.isAssignableFrom(FxNdfTrade.class)) {
+      return Optional.of(resolver.parseFxNdfTrade(baseRow, info));
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public String getName() {
+    return "FxNdf";
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Parses from the CSV row.
+   *
+   * @param row the CSV row
+   * @param info the trade info
+   * @param resolver the resolver used to parse additional information
+   * @return the parsed trade
+   */
+  static FxNdfTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    FxNdfTrade trade = parseRow(row, info);
+    return resolver.completeTrade(row, trade);
+  }
+
+  //-------------------------------------------------------------------------
+  // parses the row to a trade
+  private static FxNdfTrade parseRow(CsvRow row, TradeInfo info) {
+    CurrencyAmount settlementCurrencyNotional = CsvLoaderUtils.parseCurrencyAmountWithDirection(
+        row, SETTLEMENT_CURRENCY_FIELD, SETTLEMENT_CURRENCY_NOTIONAL_FIELD, SETTLEMENT_CURRENCY_DIRECTION_FIELD);
+    LocalDate paymentDate = row.getField(PAYMENT_DATE_FIELD, LoaderUtils::parseDate);
+    Currency settlementCurrency = row.getField(SETTLEMENT_CURRENCY_FIELD, LoaderUtils::parseCurrency);
+    Currency nonDeliverableCurrency = row.getField(NON_DELIVERABLE_CURRENCY_FIELD, LoaderUtils::parseCurrency);
+    CurrencyPair currencyPair = CurrencyPair.of(settlementCurrency, nonDeliverableCurrency);
+    FxRate agreedFxRate = FxRate.of(currencyPair, row.getField(FX_RATE_FIELD, LoaderUtils::parseDouble));
+    FxIndex index = parseFxIndex(row, currencyPair);
+    FxNdf fxNdf = FxNdf.builder()
+        .settlementCurrencyNotional(settlementCurrencyNotional)
+        .agreedFxRate(agreedFxRate)
+        .index(index)
+        .paymentDate(paymentDate)
+        .build();
+    return FxNdfTrade.of(info, fxNdf);
+  }
+
+  // parses the FX Index
+  private static FxIndex parseFxIndex(CsvRow row, CurrencyPair currencyPair) {
+    // prioritize value in fx index field if present
+    try {
+      return FxIndex.of(row.getValue(FX_INDEX_FIELD));
```

Comment:
Avoid the try-catch where possible

`FxIndex.of(row.findValue(FX_INDEX_FIELD).orElse(currencyPair.toString()))`